nip47/controllers/create_connection_controller.go

```diff
@@ -53,6 +53,8 @@ func (controller *nip47Controller) HandleCreateConnectionEvent(ctx context.Conte
 		expiresAt = &expiresAtValue
 	}
 
+	maxAmountSat := uint64(params.Budget.Budget / 1000)
```

Comment:
why is casting needed here?

Reply:
Ah I thought you have to do that while dividing, will remove :+1:

Reply:
I think go is different and will preserve the current type. The type is already a uint64 right? so it will do truncation instead

Reply:
Yeah that's good